pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -577,7 +580,22 @@ func (p *Provisioner) setAdditionalSettings(options *apibkt.BucketOptions) error
 	if err != nil {
 		return errors.Wrap(err, "failed to process additionalConfig")
 	}
+	p.additionalConfig = additionalConfig
```

Comment:
`Provisioner` is a data store that is persisted between reconciles, and adding this to it will risk reconciles of different OBCs accidentally sharing data. Let's change this to pass additional config to each `setXyzQuota()` as a parameter instead, to ensure future devs won't add or approve problem code.

Reply:
Done.